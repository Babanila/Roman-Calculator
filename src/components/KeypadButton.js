import React from "react";
import styled from "@emotion/styled";

const KeypadDiv = styled.div`
    margin-top: 1em;
    display: grid;
    background: smoke;
    grid-template-columns: repeat(4, 100px);
    z-index: 100;

    @media (min-width: 0px) and (max-width: 400px) {
        margin-top: 0.5em;
        grid-template-columns: repeat(4, 78px);
    }
`;

const Button = styled.button`
    cursor: pointer;
    font-size: 2rem;
    border: 1px solid white;
    outline: none;
    width: 3em;
    height: 3em;
    border-radius: 0.5em;
    background-color: #f6f6f6;
    &:hover {
        background: #c4c4ff;
        color: black;
    }

    @media (min-width: 0px) and (max-width: 400px) {
        width: 2em;
        height: 2em;
    }
`;

const Button2 = styled.button`
    cursor: pointer;
    font-size: 2rem;
    color: red;
    outline: none;
    height: 1.5em;
    width: 12em;
    margin: 0.1em 0.1em 0.1em 0.3em;
    border-radius: 0.5em;
    display: flex;
    justify-content: center;
    align-self: center;
    text-align: center;
    background-color: #f6f6f6;
    &:hover {
        background: #c4c4ff;
        color: black;
    }

    @media (min-width: 0px) and (max-width: 420px) {
        width: 300px;
        height: 1.2em;
    }
`;

const KeypadButton = props => {
    const keypads = [
        "I",
        "II",
        "III",
        "+",
        "IV",
        "V",
        "VI",
        "-",
        "VII",
        "VIII",
        "IX",
        "รท",
        "X",
        "L",
        "C",
        "*",
        "D",
        "M",
        "AC",
        "=",
    ];

    return (
        <KeypadDiv>
            {keypads.map((item, i) => (
                <Button
                    className="basic-Button"
                    data-label={item}
                    key={item}
                    value={item}
                    onClick={props.onClick}
                >
                    {item}
                </Button>
            ))}
            <Button2
                className="basic-Button"
                data-label="DELETE"
                key="DELETE"
                value="DELETE"
                onClick={props.onClick}
            >
                DELETE
            </Button2>
        </KeypadDiv>
    );
};

export default KeypadButton;
