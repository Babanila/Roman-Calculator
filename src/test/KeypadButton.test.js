import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import KeypadButton from "../components/KeypadButton";

describe("KeypadButton Component Test", () => {
    afterEach(cleanup);

    it("should renders without crashing", () => {
        const KeypadButtonComponent = renderer.create(<KeypadButton />).toJSON();
        expect(KeypadButtonComponent).toMatchSnapshot();
    });

    it("should render correct number of element", () => {
        const KeypadButtonWrapper = shallow(<KeypadButton />);
        expect(KeypadButtonWrapper.find("div").exists).toBeTruthy();
        expect(KeypadButtonWrapper.find("button").exists).toBeTruthy();
        expect(KeypadButtonWrapper.find("div").length).toEqual(0);
        expect(KeypadButtonWrapper.find("button").length).toEqual(0);
    });

    it("should render correct number of children element", () => {
        const KeypadButtonWrapper = mount(<KeypadButton />);
        expect(KeypadButtonWrapper.find("div").exists).toBeTruthy();
        expect(KeypadButtonWrapper.find("button").exists).toBeTruthy();
        expect(KeypadButtonWrapper.find("div").length).toEqual(1);
        expect(KeypadButtonWrapper.find("button").length).toEqual(21);
    });

    it("should show the right values for output", () => {
        const { getByText } = render(<KeypadButton />);

        expect(getByText("AC")).toBeInTheDocument();
        expect(getByText("X")).toBeInTheDocument();
        expect(getByText("L")).toBeInTheDocument();
        expect(getByText("*")).toBeInTheDocument();
    });
});
