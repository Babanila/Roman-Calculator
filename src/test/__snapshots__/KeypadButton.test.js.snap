// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeypadButton Component Test should renders without crashing 1`] = `
<div
  className="css-1u2c2og"
>
  <button
    className="basic-Button css-1kddh5u"
    data-label="I"
    value="I"
  >
    I
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="II"
    value="II"
  >
    II
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="III"
    value="III"
  >
    III
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="+"
    value="+"
  >
    +
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="IV"
    value="IV"
  >
    IV
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="V"
    value="V"
  >
    V
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="VI"
    value="VI"
  >
    VI
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="-"
    value="-"
  >
    -
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="VII"
    value="VII"
  >
    VII
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="VIII"
    value="VIII"
  >
    VIII
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="IX"
    value="IX"
  >
    IX
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="รท"
    value="รท"
  >
    รท
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="X"
    value="X"
  >
    X
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="L"
    value="L"
  >
    L
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="C"
    value="C"
  >
    C
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="*"
    value="*"
  >
    *
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="D"
    value="D"
  >
    D
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="M"
    value="M"
  >
    M
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="AC"
    value="AC"
  >
    AC
  </button>
  <button
    className="basic-Button css-1kddh5u"
    data-label="="
    value="="
  >
    =
  </button>
  <button
    className="basic-Button css-adl3hz"
    data-label="DELETE"
    value="DELETE"
  >
    DELETE
  </button>
</div>
`;
